clicache/src/Cache.cpp

```diff
@@ -57,6 +56,20 @@ namespace Apache
         m_typeRegistry = gcnew Apache::Geode::Client::TypeRegistry(this);
       }
 
+      Cache::~Cache() { //Destructor - deterministic
+        if (_disposed) return;
+        //Clean-up managed resources
+        this->!Cache();
+        _disposed = true;
+        //GC.SuppressFinalize(this) is automatically added here
+        //Base destructor is automatically called too if needed
+      }
+
+      Cache::!Cache() { //Finalizer - non-deterministic when called by GC
+        //Clean-up unmanaged resources
+        m_nativeptr->get()->~Cache();
```

Comment:
You should never call a destructor directly. Delete the smart pointer it will deleted and destruct the object.